ResearchGate.js

```diff
@@ -37,34 +37,43 @@
 
 
 // attr()/text() v2
-function attr(docOrElem,selector,attr,index){var elem=index?docOrElem.querySelectorAll(selector).item(index):docOrElem.querySelector(selector);return elem?elem.getAttribute(attr):null}function text(docOrElem,selector,index){var elem=index?docOrElem.querySelectorAll(selector).item(index):docOrElem.querySelector(selector);return elem?elem.textContent:null}
+function attr(docOrElem, selector, attr, index) {
+	var elem = index ? docOrElem.querySelectorAll(selector).item(index) : docOrElem.querySelector(selector); return elem ? elem.getAttribute(attr) : null;
+} function text(docOrElem, selector, index) {
+	var elem = index ? docOrElem.querySelectorAll(selector).item(index) : docOrElem.querySelector(selector); return elem ? elem.textContent : null;
+}
 
 function detectWeb(doc, url) {
 	if (url.includes('/publication/')) {
-		var type = text(doc, 'div.publication-meta strong');
+		var type = text(doc, "span.research-detail-header-section__badge");
 		if (!type) {
+			type = text(doc, '[data-testid="research-meta-type"]');
+			if (!type) {
 			// for logged in users (yes, really...)
-			type = text(doc, 'b[data-reactid]');
+				Zotero.debug('trying logged in way');
+				type = text(doc, 'b[data-reactid]');
+			}
 		}
-		type = type.replace('(PDF Available)', '').trim();
+		// type = type.replace('(PDF Available)', '').trim();
 		switch (type) {
-		case "Data":// until we have a data itemType
-		case "Article":
-			return "journalArticle";
-		case "Conference Paper":
-			return "conferencePaper";
-		case "Chapter":
-			return "bookSection";
-		case "Thesis":
-			return "thesis";
-		case "Research":
-			return "report";
-		case "Presentation":
-			return "presentation";
-		default:
-			return "book";
+			case "Data":// until we have a data itemType
+			case "Article":
+				return "journalArticle";
+			case "Conference Paper":
+				return "conferencePaper";
+			case "Chapter":
+				return "bookSection";
+			case "Thesis":
+				return "thesis";
+			case "Research":
+				return "report";
+			case "Presentation":
+				return "presentation";
+			default:
+				return "book";
 		}
-	} else if ((url.match('/search(\\?|/)?') || url.includes('/profile/') || url.includes('/scientific-contributions/')) && getSearchResults(doc, true)) {
+	}
+	else if ((url.match('/search(\\?|/)?') || url.includes('/profile/') || url.includes('/scientific-contributions/')) && getSearchResults(doc, true)) {
```

Comment:
```suggestion
	else if ((url.includes('/search') || url.includes('/profile/') || url.includes('/scientific-contributions/')) && getSearchResults(doc, true)) {
```

should be equivalent & much simpler. (If the regex is needed, don't use `search` or `test` to test against it)

Reply:
Fine with me. It was already like this.